core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -50,9 +54,47 @@
   private Deletes() {
   }
 
-  public static <T> CloseableIterable<T> positionFilter(CloseableIterable<T> rows, Function<T, Long> rowToPosition,
-                                                        CloseableIterable<Long> posDeletes) {
-    return new PositionDeleteFilter<>(rows, rowToPosition, posDeletes);
+  public static <T> CloseableIterable<T> equalitySetFilter(CloseableIterable<T> rows,
+                                                           Function<T, StructLike> rowToDeleteKey,
+                                                           Types.StructType eqType,
+                                                           CloseableIterable<StructLike> eqDeletes) {
+    try (CloseableIterable<StructLike> deletes = eqDeletes) {
+      CloseableIterator<StructLike> eqDeleteIterator = deletes.iterator();
+      if (eqDeleteIterator.hasNext()) {
+        StructLikeSet deleteSet = StructLikeSet.create(eqType);
+        Iterators.addAll(deleteSet, eqDeleteIterator);
```

Comment:
I am not sure I get this part. Do we create `StructLikeSet` for every entry in `eqDeletes`?

Reply:
No. The outer `hasNext` check is used to see whether we need to filter at all. If there are no equality deletes then we just return rows in the else case. This could happen if we filter the deletes using the scan predicates. If you're looking for a specific ID that is also used for a delete, then you only need to merge in the deletes with that ID.

To fill the delete set, we call `Iterators.addAll` that will add all of the remaining items from an iterator to a collection.

Reply:
Got it now.